sunpy/physics/differential_rotation.py

```diff
@@ -510,7 +510,13 @@ def differential_rotate(smap, observer=None, time=None, **diff_rot_kwargs):
 
     # Only this function needs scikit image
     from skimage import transform
-
+    # Ensure 'rsun_ref' consistency
+    rsun_ref = smap.meta.get('rsun_ref', None)
+    if rsun_ref is None:
+        raise KeyError("The input map's metadata is missing 'rsun_ref'.")
+    
+    # Use 'rsun_ref' throughout the function
+    rsun_ref = rsun_ref * u.m
```

Comment:
Do we need all of this?

Reply:
Just for error handling..??

Reply:
I guess my actual question is, how does this help solve the original issue? 

It does make the error clearer to users than the current behavior.
I would want some other input but having the code not error but raise a warning saying its missing and using the default value used elsewhere in sunpy seems like the simplest answer.

Reply:
Ok got it, will change and update the error to warning and update for default nature.